src/bin/tikv-server.rs

```diff
@@ -180,27 +180,22 @@ fn check_system_config(config: &toml::Value) {
     }
 }
 
-fn listen_address(matches: &Matches, config: &toml::Value) -> String {
-    let addr = get_flag_string(matches, "A").unwrap_or_else(|| {
-        get_toml_string(config,
-                        "server.addr",
-                        Some(DEFAULT_LISTENING_ADDR.to_owned()))
-    });
-    util::config::check_addr(&addr)
+fn check_advertise_address(addr: &str) {
+    util::config::check_addr(addr)
         .map_err(|e| exit_with_err(format!("{:?}", e)))
         .unwrap();
 
-    let adv_addr = get_flag_string(matches, "advertise-addr")
-        .unwrap_or_else(|| get_toml_string(config, "server.advertise-addr", Some(addr.clone())));
-    util::config::check_addr(&adv_addr)
-        .map_err(|e| exit_with_err(format!("{:?}", e)))
-        .unwrap();
+    // FIXME: Forbidden addresses ending in 0 or 255? Those are not always invalid.
+    // See more: https://en.wikipedia.org/wiki/IPv4#Addresses_ending_in_0_or_255
+    let invalid_patterns = [("0.", 0)]; // Current network is not allowed.
 
-    if let Some(_) = adv_addr.find("0.0.0.0") {
-        exit_with_err("0.0.0.0 is not allowed in advertise-addr".to_owned());
+    for &(pat, pos) in &invalid_patterns {
```

Comment:
I think `("0.", 0)` is okay, it is more flexible than `0.0.0.0`, plus `0.*.*.*` are definitely invalid.